Distribution/Server/Features/Users.hs

```diff
@@ -468,6 +493,66 @@ userFeature  usersState adminsState
           errBadRequest "User deleted"
             [MText "Cannot disable account, it has already been deleted"]
 
+    serveUserManagement :: DynamicPath -> ServerPartE Response
+    serveUserManagement dpath =
+        do (UserName username) <- userNameInPath dpath
+           uid <- lookupUserName (UserName username)
+           guardAuthorised [IsUserId uid, InGroup adminGroup]
+           userInfo <- lookupUserInfo uid
+           let (UserTokenMap knownTokens) = userTokens userInfo
+           template <- getTemplate templates "manage.html"
+           let mkTok (t, desc) =
+                   templateDict
+                   [ templateVal "hash" (display t)
+                   , templateVal "description" desc
+                   ]
+           ok $ toResponse $ template
+               [ "username" $= username
+               , "tokens" $= map mkTok (Map.toList knownTokens)
+               ]
+
+    runUserManagementAction :: DynamicPath -> ServerPartE Response
+    runUserManagementAction dpath =
+        do (UserName username) <- userNameInPath dpath
+           uid <- lookupUserName (UserName username)
+           guardAuthorised [IsUserId uid, InGroup adminGroup]
+           action <- look "action"
+           case action of
+             "new-auth-key" ->
```

Comment:
Style is to keep the 'do' on the previous line

    blah -> do
        ... stuff